### PR TITLE
[ecs] Update fields.ecs.yml to 1.12

### LINKED ISSUE #27835: Upgrade ECS to 1.12
**Describe the enhancement:**
All beats and modules need to be upgraded to use the ECS 1.12 version.

- [x] fields.ecs.yml is updated
- [x] beats are updated
- [x] modules are updated

### PR DESCRIPTION
**Pull request by marc-gr**
## What does this PR do?

- [x] fields.ecs.yml to ECS 1.12.0-dev
- [x] bump ecs.version in configs for modules that don't require changes.

## Why is it important?

- keep up to date with ECS changes
- allows modules to use new ECS fields

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

```
cd filebeat && mage pythonIntegTest
cd x-pack/filebeat && mage pythonIntegTest
```

Closes #27835

### CODE REVIEW

**Review by andrewkroh — CHANGES_REQUESTED** [review]
This changes `message` to use `match_only_text` which was introduced in 7.14. I think libbeat should fall-back to `text` if that type is not available. We added something similar for `wildcard` at https://github.com/elastic/beats/blob/7a49ca15950fed297aac20d3d96fe605db65164b/libbeat/template/processor.go#L90-L96.

**Review by leehinman — APPROVED** [review]
LGTM

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -506,6 +506,8 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Add Huawei Cloud provider to add_cloud_metadata. {pull}27607[27607]
 - Add default seccomp policy for linux arm64. {pull}27955[27955]
 - Add cluster level add_kubernetes_metadata support for centralized enrichment {pull}24621[24621]
+- Update ECS to 1.12.0. {pull}27770[27770]
+- Add fallback for `match_only_text`. {pull}27770[27770]
```
Comment (andrewkroh) [inline CHANGELOG.next.asciidoc:510]: Could you add a bit more description here. Like "Fields mapped as `match_only_text` will automatically fallback to a `text` mapping when using Elasticsearch versions that do not support `match_only_text`."

**Review by andrewkroh — APPROVED** [review]

**Review by mdelapenya — APPROVED** [review]
From E2E stand point, I'd say this PR is OK, as the errors we are seeing in the build history are considered flaky

**Pull request merged**